File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solution(self,root,ans):
        if(root == None):
            return 0;
        ans = (ans*10)+root.val
        if(root.left==None and root.right==None):
            return ans
        return self.solution(root.left,ans)+self.solution(root.right,ans)

    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        ans=self.solution(root,0)
        return ans
        

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int solve(TreeNode* root,int totalsum){
        if(!root) return 0;
        totalsum=totalsum*10+root->val;
        if(!root->left && !root->right){
            return totalsum;
        }
        int leftsum=solve(root->left,totalsum);
        int rightsum=solve(root->right,totalsum);
        return leftsum+rightsum;
    }
    int sumNumbers(TreeNode* root) {
       return solve(root,0);
    }
};